geps/gep-3155/index.md

```diff
@@ -0,0 +1,264 @@
+policies to selectively target specific Gateway instances, GatewayClasses or gateway 
+namespaces.
+
+### SANs on BackendTLSPolicy
+
+This change enables the certificate to have a different identity than the SNI
+(both are currently tied to the hostname field). This is particularly useful
+when using SPIFFE, which relies on URI Subject Names which are not valid SNIs 
+as per https://www.rfc-editor.org/rfc/rfc6066.html#section-3.
+
+In such case either connection properties or an arbitrary SNI, like cluster-local 
+service name could be used for certificate selection, while the identity validation
+will be done based on SubjectAltNames field.
+
+**1. Add a new `SubjectAltNames` field to `BackendTLSPolicyValidation`**
+
+```go
+type BackendTLSPolicyValidation struct {
+  // SubjectAltNames contains one or more Subject Alternative Names.
+  // When specified, at least one of certificate's Subject Alternate Names MUST 
+  // match at least one of the specified SubjectAltNames.
+  // +kubebuilder:validation:MaxItems=5
+  SubjectAltNames []SubjectAltName `json:"subjectAltNames,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Cookie;Header
+type SubjectAltNameType string
+
+const (
+   // HostnameSubjectAltNameType specifies hostname-based SAN.
+   //
+   // Support: Core
+   HostnameSubjectAltNameType SubjectAltNameType = "Hostname"
+
+   // URISubjectAltNameType specifies URI-based SAN, e.g. SPIFFE id.
+   //
+   // Support: Core
+   URISubjectAltNameType SubjectAltNameType = "URI"
+)
+
+
+type SubjectAltName struct {
+  // Type determines the format of the Subject Alternative Name. Always required.
+  Type SubjectAltNameType `json:"type"`
+
+  // Hostname contains Subject Alternative Name specified in DNS name format. Required when Type is set to Hostname, ignored otherwise.
+  Hostname v1.PreciseHostname `json:"hostname,omitempty"`
+
+  // URI contains Subject Alternative Name specified in URI format. Required when Type is set to URI, ignored otherwise.
+  URI string `json:"uri,omitempty"`
+}
+```
+
+**2. Modify Spec for `BackendTLSPolicyValidation` `Hostname`**
+
+Before:
+```go
+  // 2. Hostname MUST be used for authentication and MUST match the certificate
+  //    served by the matching backend.
+```
+
+After:
+```go
+  // 2. Only if SubjectAltNames is not specified, Hostname MUST be used for 
+  //    authentication and MUST match the certificate served by the matching
+  //    backend.
+```
+
+### Allow per-service mTLS settings BackendTLSPolicy
+
+Gateway level TLS configuration already includes an `options` field. This has
+been helpful for implementation-specific TLS configurations, or simply features
+that have not made it to the core API yet. It would be similarly useful to have
+an identical field on BackendTLSPolicy.
+
+Examples:
+- configuration options for vendor-specific mTLS automation
+- restrictions on the minimum supported TLS version or supported cipher suites
```

Comment:
I just want to be on record as saying that I didn't like this field when we added it to Gateway, I don't like this field, but I don't have a good enough reason to block any changes. Let's see if it ends up misused in the wild.

Reply:
I definitely agree with the hesitation here, but ultimately believe this is more helpful than not. There's such a wide range of TLS config possible that it seems unlikely that we'll ever cover all of the potential concepts in the core API. I think it's likely preferable to mirror the options map we have on Gateway Listeners than to leave out a form of extensibility here, though I'll admit that both options have risks.